extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetSchemaReader.java

```diff
@@ -143,18 +147,35 @@ public static ParquetInstructions readParquetSchema(
         };
         final ParquetMessageDefinition colDef = new ParquetMessageDefinition();
         final Map<String, String[]> parquetColumnNameToFirstPath = new HashMap<>();
-        for (final ColumnDescriptor column : schema.getColumns()) {
+        final Map<Integer, Long> fieldIdCount = schema.getColumns()
+                .stream()
+                .map(ColumnDescriptor::getPrimitiveType)
+                .map(Type::getId)
+                .filter(Objects::nonNull)
+                .map(ID::intValue)
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        if (schema.getFieldCount() != schema.getColumns().size()) {
+            throw new IllegalStateException(String.format(
+                    "Field count inconsistent with number of columns, schema.getFieldCount()=%d, schema.getColumns().size()=%d",
+                    schema.getFieldCount(), schema.getColumns().size()));
```

Comment:
Do we really think this holds? I wonder, since "field" and "column" are distinct names.